### PR TITLE
Dev magda connect gui with client

### PR DESCRIPTION
**Pull request by M-Aarsman**

### CODE REVIEW

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    private int port;
+    private final Socket clientSocket;
+
+    private static final Target logger = new SharedLogger(Client.class);
+
+    public Client() {
+        this.clientSocket = new Socket();
+    }
+
+    public boolean connect(String ipAddress, int port) {
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: final to parameters

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        try {
+            address = InetAddress.getByName(this.ipAddress);
+        } catch (UnknownHostException e) {
+            logger.error(e.getMessage());
+            return false;
+        }
+
+        try {
+            clientSocket.connect(new InetSocketAddress(address, this.port));
+        } catch (IOException e) {
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Pipe

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    private static final Target logger = new SharedLogger(Client.class);
+
+    public Client() {
+        this.clientSocket = new Socket();
+    }
+
+    public boolean connect(String ipAddress, int port) {
+        this.ipAddress = ipAddress;
+        this.port = port;
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Are those two used anywhere apart from this method?

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        InetAddress address = null;
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Do we need to do it outside of try?

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        } catch (IllegalArgumentException e) {
+            logger.error(e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public void run() {
+        Sender sender = null;
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Do we need to do it outside of try?

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+        //First Message
+            Message firstMessage = new MessageBuilder()
+                    .withHeader("Connection")
+                    .withStatus("OK")
+                    .withAuthor("Magda")
+                    .withStatement("Hey, it's Magda :-)")
+                    .build();
+            sender.send(firstMessage);
+        Receiver receiver = null;
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Do we need to do it outside of try?

File: client/src/main/java/com/epam/ships/client/client/Client.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        Sender sender = null;
+        try {
+            sender = new JSONSender(clientSocket.getOutputStream());
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+            //TODO: handle
+        }
+
+        //First Message
+            Message firstMessage = new MessageBuilder()
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Maybe move to try with Sender

Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/client/Client.java]: Maybe move to try where Sender is instantiated

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        imPolandFlag.setOpacity(imageCapacity);
+        imEnglandFlag.setOpacity(imageCapacity);
+        imCannon.setOpacity(imageCapacity);
+        vbWheel.setVisible(true);
+
+        String serverAddress = tServerAddress.getText();
+
+        logger.info("server address: " + serverAddress);
+
+        int serverPort = Integer.parseInt(tServerPort.getText());
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: Integer.valueOf

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                String serverNotRespondingURL = "/fxml/serverNotResponse.fxml";
+                FXMLLoader notResponseLoader = new FXMLLoader(getClass().getResource(serverNotRespondingURL));
+                Parent notResponse = notResponseLoader.load();
+                Pane mainPane = (Pane) mainAnchorPane.getParent();
+
+                mainPane.getChildren().clear();
+                mainPane.getChildren().setAll(notResponse);
+
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: logger.error

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            }
+        } else {
+            Thread clientThread = new Thread(client);
+            clientThread.start();
+        }
+    }
+
+    void initialize(ImageView imPolandFlag, ImageView imEnglandFlag, Client client) {
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: final to arguments

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    private void onConnectPressed() {
+        final double imageCapacity = 0.4;
+
+        gridPane.setDisable(true);
+        imPolandFlag.setOpacity(imageCapacity);
+        imEnglandFlag.setOpacity(imageCapacity);
+        imCannon.setOpacity(imageCapacity);
+        vbWheel.setVisible(true);
+
+        String serverAddress = tServerAddress.getText();
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: final

File: client/src/main/java/com/epam/ships/client/gui/MainController.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+    @FXML
+    private ImageView imPolandFlag;
+
+    @FXML
+    private ImageView imEnglandFlag;
+
+    @FXML
+    void initialize(final Client client) {
+        String connectWindowURL = "/fxml/connectWindow.fxml";
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/MainController.java]: final

File: client/src/main/java/com/epam/ships/client/gui/MainController.java [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+        try {
+            FXMLLoader connectLoader = new FXMLLoader(getClass().getResource(connectWindowURL));
+            Parent connect = connectLoader.load();
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/MainController.java]: final to both

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java (hunk 8 above)
Comment (korotkevics) [reply client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: final

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+        int serverPort = Integer.parseInt(tServerPort.getText());
+
+        logger.info("server port: " + serverPort);
+
+        //TODO:check if client != null
+
+        boolean isConnect = client.connect(serverAddress, serverPort);
+
+        if(!isConnect) {
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: isConnected

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+
+        logger.info("server address: " + serverAddress);
+
+        int serverPort = Integer.parseInt(tServerPort.getText());
+
+        logger.info("server port: " + serverPort);
+
+        //TODO:check if client != null
+
+        boolean isConnect = client.connect(serverAddress, serverPort);
```
Comment (korotkevics) [inline client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: final, isConnected

File: pom.xml [hunk 16]
```diff
@@ -64,6 +64,7 @@
         <orgJsonVersion>20171018</orgJsonVersion>
         <commonsIoVersion>2.5</commonsIoVersion>
         <jacksonVersion>2.9.3</jacksonVersion>
+        <lombokVersion>1.16.16</lombokVersion>
```
Comment (korotkevics) [inline pom.xml]: Why not 1.16.18.1?

File: client/src/main/java/com/epam/ships/client/client/Client.java (hunk 1 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/client/Client.java]: Done

File: client/src/main/java/com/epam/ships/client/client/Client.java (hunk 3 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/client/Client.java]: Done

File: client/src/main/java/com/epam/ships/client/client/Client.java (hunk 4 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/client/Client.java]: ok

File: client/src/main/java/com/epam/ships/client/client/Client.java (hunk 7 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/client/Client.java]: ok

File: client/src/main/java/com/epam/ships/client/client/Client.java (hunk 5 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/client/Client.java]: ok

File: client/src/main/java/com/epam/ships/client/gui/MainController.java (hunk 12 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/gui/MainController.java]: ok

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java (hunk 15 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: ok

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java (hunk 8 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: ok

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java (hunk 9 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: ok

File: client/src/main/java/com/epam/ships/client/gui/StartWindowController.java (hunk 10 above)
Comment (M-Aarsman) [reply client/src/main/java/com/epam/ships/client/gui/StartWindowController.java]: ok

**Pull request merged**